src/main/java/org/alliancegenome/curation_api/dao/associations/alleleAssociations/AlleleGeneAssociationDAO.java

```diff
@@ -0,0 +1,24 @@
+package org.alliancegenome.curation_api.dao.associations.alleleAssociations;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.Query;
+
+import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
+import org.alliancegenome.curation_api.model.entities.associations.alleleAssociations.AlleleGeneAssociation;
+
+@ApplicationScoped
+public class AlleleGeneAssociationDAO extends BaseSQLDAO<AlleleGeneAssociation> {
+
+	protected AlleleGeneAssociationDAO() {
+		super(AlleleGeneAssociation.class);
+	}
+
+	public List<Long> findAllIdsByDataProvider(String dataProvider) {
+		Query jpqlQuery = entityManager.createQuery("SELECT aga.id FROM AlleleGeneAssociation aga WHERE aga.subject.dataProvider.sourceOrganization.abbreviation = :dataProvider");
```

Comment:
As a point for the future we need to create a generic method that does this.

Reply:
Yeah, I'll have a think about that.  It's slightly different for different classes, but there's a lot of duplication and will be more so once we start adding more association classes.

Reply:
Ya we could have a method that is like the other one where you pass in the params but then you could also pass in the "id" field then all the same code could be run...